src/components/common/SideContent.js

```diff
@@ -0,0 +1,195 @@
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { makeStyles, useTheme } from '@material-ui/core/styles';
+import IconButton from '@material-ui/core/IconButton';
+import ForumIcon from '@material-ui/icons/Forum';
+import PushPinIcon from '@material-ui/icons/PushPin';
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { Box, Paper, Slide } from '@material-ui/core';
+import { Map } from 'immutable';
+import { useTranslation } from 'react-i18next';
+import Chatbox from './Chatbox';
+import { HEADER_HEIGHT, DRAWER_WIDTH } from '../../config/constants';
+import { LayoutContext } from '../context/LayoutContext';
+import Item from './Item';
+import {
+  ITEM_CHATBOX_BUTTON_ID,
+  PANNEL_CLOSE_BUTTON_ID,
+  ITEM_PINNED_BUTTON_ID,
+  ITEM_PINNED_ID,
+} from '../../config/selectors';
+import { getDirectParentId } from '../../utils/item';
+import { ITEM_TYPES } from '../../enums';
+
+const useStyles = makeStyles((theme) => ({
+  iconButton: {
+    float: 'right',
+  },
+  root: {
+    display: 'flex',
+  },
+  drawer: {
+    width: DRAWER_WIDTH,
+    flexShrink: 0,
+  },
+  drawerHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: theme.spacing(0, 1),
+    // necessary for content to be below app bar
+    justifyContent: 'flex-start',
+  },
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+  },
+  contentShift: {
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginRight: 0,
+  },
+}));
+
+const SideContent = ({ children, item }) => {
+  const parentId =
+    (item.get('type') !== ITEM_TYPES.FOLDER &&
+      getDirectParentId(item.get('path'))) ||
+    item.get('id');
+  const settings = item.get('settings');
+
+  const {
+    isPinnedMenuOpen,
+    setIsPinnedMenuOpen,
+    isChatboxMenuOpen,
+    setIsChatboxMenuOpen,
+  } = useContext(LayoutContext);
+
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const theme = useTheme();
+
+  const toggleChatOpen = () => {
+    setIsChatboxMenuOpen(!isChatboxMenuOpen);
+    setIsPinnedMenuOpen(false);
+  };
+
+  const togglePinnedOpen = () => {
+    setIsPinnedMenuOpen(!isPinnedMenuOpen);
+    setIsChatboxMenuOpen(false);
+  };
+
+  const diplayChatButton = () => {
+    if (!settings?.showChatbox) return null;
+
+    return (
+      <IconButton
+        id={ITEM_CHATBOX_BUTTON_ID}
+        className={classes.iconButton}
+        aria-label={isChatboxMenuOpen ? t('Hide Chat') : t('Show Chat')}
+        onClick={toggleChatOpen}
+      >
+        <ForumIcon />
+      </IconButton>
+    );
+  };
+
+  const displayChatbox = () => {
```

Comment:
If possible, I think it makes sense to create whole component for this/these components.